syntax = "proto3";

package stock;

message StockQuote {
    string ts_code = 1;           // 股票代码
    int64 timestamp = 2;          // 时间戳
    double last_price = 3;        // 最新价
    double open_price = 4;        // 开盘价
    double high_price = 5;        // 最高价
    double low_price = 6;         // 最低价
    double pre_close = 7;         // 昨收价
    int64 volume = 8;             // 成交量
    double amount = 9;            // 成交额
    int64 pvolume = 10;          // 持仓量
    int32 transaction_num = 11;   // 成交笔数
    int32 stock_status = 12;      // 股票状态

    // 买一到买五档
    double bid_price1 = 13;
    int64 bid_volume1 = 14;
    double bid_price2 = 15;
    int64 bid_volume2 = 16;
    double bid_price3 = 17;
    int64 bid_volume3 = 18;
    double bid_price4 = 19;
    int64 bid_volume4 = 20;
    double bid_price5 = 21;
    int64 bid_volume5 = 22;

    // 卖一到卖五档
    double ask_price1 = 23;
    int64 ask_volume1 = 24;
    double ask_price2 = 25;
    int64 ask_volume2 = 26;
    double ask_price3 = 27;
    int64 ask_volume3 = 28;
    double ask_price4 = 29;
    int64 ask_volume4 = 30;
    double ask_price5 = 31;
    int64 ask_volume5 = 32;
}

message StockQuoteBatch {
    int64 batch_timestamp = 1;     // 批次时间戳
    string publisher_id = 2;       // 发布者ID
    repeated StockQuote quotes = 3; // 行情数据列表
    string message_type = 4;       // 消息类型: "MARKET_DATA" 或 "HEARTBEAT"
}